app/javascript/app/components/emission-pathways-overview/emission-pathways-overview-selectors.js

```diff
@@ -0,0 +1,51 @@
+import { createSelector } from 'reselect';
+import pick from 'lodash/pick';
+import isEmpty from 'lodash/isEmpty';
+
+const getId = stateWithId => stateWithId.id || null;
+const getCategory = stateWithId => stateWithId.category || null;
+const getCategoryData = stateWithId => stateWithId.categoryData.data || null;
+
+const getOverviewData = createSelector([getCategoryData, getId], (data, id) => {
+  if (isEmpty(data)) return null;
+  const idData = data.find(d => String(d.id) === id);
+  if (!idData) return null;
+  return idData;
+});
+
+const selectOverviewData = createSelector(
+  [getOverviewData, getCategory],
+  (data, category) => {
+    const overviewFields = {
+      Models: [
+        'maintainer_name',
+        'geographic_coverage_region',
+        'sectoral_coverage',
+        'time_horizon',
+        'license'
+      ],
+      Scenarios: [],
+      Indicators: []
+    };
+    return pick(data, overviewFields[category]);
+  }
+);
+
+export const parseArraysOverviewData = createSelector(
+  [selectOverviewData],
+  data => {
+    const parsedData = {};
+    Object.keys(data).forEach(key => {
+      let fieldData = data[key];
+      if (fieldData && typeof fieldData !== 'string') {
```

Comment:
What happens if it isn't string or array?

Reply:
I think is too much checking for all the types. If someone puts there an object It should break. I have written some doc before the function. Or then we should try typescript or flow 😄